brave/src/main/java/zipkin2/reporter/brave/ZipkinSpanHandler.java

```diff
@@ -53,28 +55,40 @@ public static SpanHandler create(Reporter<Span> spanReporter) {
   /** @since 2.13 */
   public static Builder newBuilder(Reporter<Span> spanReporter) {
     if (spanReporter == null) throw new NullPointerException("spanReporter == null");
-    return new ConvertingSpanReporterBuilder(spanReporter);
+    return new ConvertingZipkinSpanHandler.Builder(spanReporter);
   }
 
-  static final class ConvertingSpanReporterBuilder extends Builder {
-    final Reporter<Span> spanReporter;
-
-    ConvertingSpanReporterBuilder(Reporter<Span> spanReporter) {
-      this.spanReporter = spanReporter;
-    }
+  /**
+   * Allows this instance to be reconfigured, for example {@link Builder#alwaysReportSpans(boolean)}.
+   *
+   * <p><em>Note:</em>If the instance  {@link AsyncReporter#close()} if you no longer need this
+   * instance, as otherwise it can leak its reporting thread.
+   *
+   * @since 2.15
+   */
+  public Builder toBuilder() {
+    // For testing, this is easier than making the type abstract: It is package sealed anyway!
+    throw new UnsupportedOperationException();
+  }
 
-    // SpanHandler not ZipkinSpanHandler as it can coerce to NOOP
-    @Override public SpanHandler build() {
-      if (spanReporter == Reporter.NOOP) return SpanHandler.NOOP;
-      return new ZipkinSpanHandler(new ConvertingSpanReporter(spanReporter, errorTag),
-          alwaysReportSpans);
-    }
+  /**
+   * Implementations that throw exceptions on close have bugs. This may result in log warnings,
+   * though.
+   *
+   * @since 2.15
+   */
+  @Override public void close() {
```

Comment:
without this, the caller will have to play games to figure out if it should be closed or not. This lets us get the wildcard "Closeable" like we do with zipkin2.Component, but without having to couple this type to zipkin2 package.